### PR TITLE
HBASE-23615 Use a dedicated thread for executing WorkerMonitor in Pro…

### PR DESCRIPTION
**Pull request by binlijin**
…cedureExecutor.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 19s |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 18s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 41s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 22s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  21m  0s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 48s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 52s |  hbase-procedure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 13s |  The patch does not generate ASF License warnings.  |
|  |   |  64m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/961 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0871e23f4c41 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-961/out/precommit/personality/provided.sh |
| git revision | master / fc15ea7546 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/1/testReport/ |
| Max. process+thread count | 270 (vs. ulimit of 10000) |
| modules | C: hbase-procedure U: hbase-procedure |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java [hunk 1]
```diff
@@ -217,6 +217,11 @@ protected boolean shouldKillAfterStoreUpdate(final boolean isSuspended) {
    */
   private TimeoutExecutorThread<TEnvironment> timeoutExecutor;
 
+  /**
+   * Use a dedicated thread for executing WorkerMonitor.
```
Comment (Apache9) [inline hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java]: Add more comments here to explain why we need a dedicated thread?

Comment (binlijin) [reply hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java]: Ok, done.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 48s |  master passed  |
| +1 :green_heart: |  compile  |   1m 15s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 44s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 47s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 33s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 34s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 33s |  hbase-procedure in the patch passed.  |
| +1 :green_heart: |  unit  | 155m  0s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 49s |  The patch does not generate ASF License warnings.  |
|  |   | 226m  6s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/961 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux e8ed75d090e7 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-961/out/precommit/personality/provided.sh |
| git revision | master / 923ba7763e |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/2/testReport/ |
| Max. process+thread count | 4407 (vs. ulimit of 10000) |
| modules | C: hbase-procedure hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-961/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by binlijin** [comment]
@saintstack @Apache9 @virajjasani  Thanks for the review.

**Pull request merged**